### PR TITLE
AA-13801 Create `no-declare` rule to prevent using `declare` keyword

### PR DESCRIPTION
**Pull request by norbertprocak**
Jira: https://livechatinc.atlassian.net/browse/AA-13801

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@livechat/eslint-plugin",
+  "version": "0.1.7",
```
Comment (dawid-drelichowski) [inline package.json]: Why such a strange version? Shouldn't we start with 1.0.0?

File: src/rules/no-declare.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  defaultOptions: [
+    {
+      onlyIdentifiers: [],
+      excludeIdentifiers: [],
+    },
+  ],
+  create(context, [options]) {
+    const { onlyIdentifiers, excludeIdentifiers } = options;
+    const onlyIdentifiersEnabled =
+      onlyIdentifiers && onlyIdentifiers?.length > 0;
```
Comment (AlithAnar) [inline src/rules/no-declare.ts]:
```suggestion
      !!onlyIdentifiers?.length;
```

File: src/rules/no-declare.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    const excludeIdentifiersEnabled =
+      excludeIdentifiers && excludeIdentifiers.length > 0;
```
Comment (AlithAnar) [inline src/rules/no-declare.ts]: !!excludeIdentifiers.length

File: src/rules/no-declare.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    const noIdentifiersSpecified =
+      !onlyIdentifiersEnabled && !excludeIdentifiersEnabled;
+
+    return {
+      VariableDeclaration(node) {
+        if (node.declare) {
```
Comment (AlithAnar) [inline src/rules/no-declare.ts]:
```suggestion
        if (!node.declare) {
           return
        }
```

**Review by mreszke — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (norbertprocak) [reply package.json]: it's my working version, I will replace it with 1.0.0 when task is ready to merge

File: src/rules/no-declare.ts (hunk 2 above)
Comment (norbertprocak) [reply src/rules/no-declare.ts]: done 😆

File: src/rules/no-declare.ts (hunk 3 above)
Comment (norbertprocak) [reply src/rules/no-declare.ts]: done 😛

File: src/rules/no-declare.ts (hunk 4 above)
Comment (norbertprocak) [reply src/rules/no-declare.ts]: ok

**Review by dawid-drelichowski — APPROVED** [review]

**Review by AlithAnar — APPROVED** [review]

**Pull request merged**